es6-shim.js

```diff
@@ -291,6 +291,13 @@
       return x != null && Object(x) === x;
     },
 
+    TypeIsNotObject: function (val) {
```

Comment:
http://www.ecma-international.org/ecma-262/6.0/#sec-ecmascript-data-types-and-values actually only defines the `Type` operation and then the spec says things like `if Type(x) is Object, then..`.  There is no mention of `IsPrimitive` in the spec text.

On the other hand, the `TypeIsNotObject` form actually seems to be faster even when the argument is an Object.  So perhaps I should just combine these two cases and use the `TypeIsNotObject` implementation always.  I'd originally preferred the `TypeIsObject` version because it corresponded more closely to the spec text, but that correspondence seems to be gone in the latest spec version.